register.go

```diff
@@ -87,6 +87,13 @@ func roundTripAdapter(next http.RoundTripper) http.RoundTripper {
 			return nil, errors.New("unix transport: invalid path")
 		}
 
+		// abstract sockets are prefixed with an @ character
+		// but this information is lost when the URL is parsed
+		// so we need to re-add it here
+		if req.URL.String()[len(req.URL.Scheme)+3] == '@' {
```

Comment:
What are you trying to detect here? Is it that the first character in the host portion of the URL is `@`?

Reply:
Yes, exactly.
Basically the url would be written like this:
- `http+unix://@/non/existing/path/to/socket:/foo/bar`
- `https+unix://@some-unique-ident:/foo/bar`
- etc.